### ISSUE TITLE
Poll for image processing when returning a 423

### ISSUE THREAD

**Issue by colbyfayock** [colbyfayock/next-cloudinary#1]
When using transformations that require processing thus returning a 423, poll for that image on the client so it loads upon completion.

**Comment by colbyfayock** [comment]
This can potentially use the `onError` function available in next/future/image: https://nextjs.org/docs/api-reference/next/future/image#onerror

**Comment by michizhou** [comment]
Hi @colbyfayock, I would like to take on this issue.

**Comment by Smartmind12** [comment]
Hey @colbyfayock I too have tried my hands on this issue...so can i submit a PR for it?

**Comment by colbyfayock** [comment]
@michizhou was first to ask so they're first to give a try. @Smartmind12 if they dont end up working on it you can give it a shot

**Comment by Smartmind12** [comment]
> @michizhou was first to ask so they're first to give a try. @Smartmind12 if they dont end up working on it you can give it a shot

Okay thanks! 
Requesting you to kindly open up some new issues to work on in the meantime😊

**Comment by colbyfayock** [comment]
@Smartmind12 thanks for your interest, I'll try to add some more, in the meantime feel free to open any you think could be valuable, especially in terms of bringing new Cloudinary features into the library

**Comment by michizhou** [comment]
@colbyfayock Could you point out which files I should modify for this issue? I have a rough idea on how to resolve this issue but I'm not certain where a 423 error code would be returned when using transformations that need processing.

**Comment by colbyfayock** [comment]
I have two thought processes around this

**my original intent** was to use a `ref` on the `Image` inside of here: 

https://github.com/colbyfayock/next-cloudinary/blob/main/next-cloudinary/src/components/CldImage/CldImage.js#L60

where we could set an `onerror` handler: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#image_loading_errors

such as what I'm doing here:

https://github.com/colbyfayock/cloudycam/blob/4293c0551ebc64b97f221a13cccf6c4024e15442/src/components/Camera/Camera.js#L38

but instead of `onload` like in that example, use `onerror`

**howver, Next.js is releasing a new `onError` prop** https://nextjs.org/docs/api-reference/next/future/image#onerror

So it might make the most sense to develop this for the new Image component which will be launched I believe at Next.js Conf at the end of this month

Here's a PR i made that updates the component to use Future Image: https://github.com/colbyfayock/next-cloudinary/pull/3

**as far as polling goes** I haven't tested to see if they return an error object with that information, so what we could potentially do is when an error occurs, make a simple `fetch` to the image URL, check THAT response, and if that's a 423, continue polling until it's available

what do you think?

**Comment by michizhou** [comment]
I would think the second approach probably makes more sense in terms of being up to date with the upcoming release. Just to confirm, the new `onError` prop and new Image component are available in Future Image right? If not, we can go with your original intent.

**Comment by colbyfayock** [comment]
yup they're available in Future Image

that PR has all the changes I needed to make

if you're familiar with how to do it, you can create a new branch from the Future branch and target that so that we can merge your PR into the Future PR, then once Future is released, I'll update the import and merge it all in

i would think it would still count for hacktoberfest for you given it's a merged PR, if not we can figure something out to get you credit

would that work?

**Comment by michizhou** [comment]
Yeah that sounds good! I'll create a PR on top of yours and we can go from there.

**Comment by michizhou** [comment]
@colbyfayock I have pushed a PR with initial changes on top of your Future PR. It's in an early stage, and I'm not sure if it's using a proper approach, so I'd appreciate it if you could review it and let me know what additional changes are needed. In the meantime, could you reopen this issue? I think another PR you recently merged accidentally closed this.

**Comment by colbyfayock** [comment]
oops, reopened! i'll take a look :)

**Comment by michizhou** [comment]
@colbyfayock I just noticed that you merged in the feature branch I based my PR on, which closed it. I'll rebase my feature branch to be on top of the main branch and open a new PR.

**Comment by colbyfayock** [comment]
ohhhh sorry about that - please let me know if there are any issues or conflicts, there were a lot of big  changes, i can help address the conflicts if needed

**Comment by michizhou** [comment]
No worries! I'll let you know if I run into issues.

**Comment by colbyfayock** [comment]
@michizhou did you ever get this running locally? I just wrote some docs to improve helping people get set up, including [writing a script that will upload images](https://github.com/colbyfayock/next-cloudinary/pull/50) to your Cloudinary account for running the Docs site

if you haven't would love to see you go through that to make sure i didnt miss anything and to validate its usefulness

https://github.com/colbyfayock/next-cloudinary#-contributing

thanks

**Comment by michizhou** [comment]
@colbyfayock I haven't yet, but I would love to go through the docs and validate them as I set up!

**Comment by michizhou** [comment]
@colbyfayock It seems you accidentally closed this issue again with another PR you recently merged. Please reopen this issue whenever you can. In the meantime, perhaps the PR template should be modified until this is actually resolved?

**Comment by colbyfayock** [comment]
what's actually happening that this keeps getting closed? if you know the issue, mind submitting a PR to fix it?

**Comment by michizhou** [comment]
@colbyfayock I have made changes to my latest PR that resolves this issue and also submitted a new PR which fixes the bug that causes this issue to keep getting closed. Please feel free to review them both whenever you are available.

I also went through the docs which you introduced [here](https://github.com/colbyfayock/next-cloudinary/pull/50) to help people get set up with running the Docs site and uploading example images. Your changes look good and were very useful, especially when setting up my local environment!
